src/containers/language-selector.jsx

```diff
@@ -36,16 +39,19 @@ class LanguageSelector extends React.Component {
 
 LanguageSelector.propTypes = {
     children: PropTypes.node,
+    currentLocale: PropTypes.string.isRequired,
+    locales: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
     onChangeLanguage: PropTypes.func.isRequired
 };
 
 const mapStateToProps = state => ({
-    currentLocale: state.intl.locale
+    currentLocale: state.scratchGui.locales.locale,
+    locales: state.scratchGui.locales.messages
```

Comment:
Looking at https://github.com/chrisgarrity/scratch-gui/blob/a04554229f9311595c5d1624839f01bf1c6a8dc7/src/components/language-selector/language-selector.jsx#L29, it doesn't look like we even need to pass this in, since it's taken directly from `scratch-l10n`?  It doesn't seem right that the value of `locales` is set to `messages`.